crates/shielder-relayer/src/price_feed.rs

```diff
@@ -0,0 +1,125 @@
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+use rust_decimal::Decimal;
+use serde::Deserialize;
+use strum::IntoEnumIterator as _;
+use strum_macros::EnumIter;
+use time::OffsetDateTime;
+use tokio::time::Duration;
+
+const BASE_PATH: &str = "https://api.diadata.org/v1/assetQuotation";
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash, EnumIter)]
+pub enum Coin {
```

Comment:
I suggest adding `Custom(&'static str)` variant, so that anyone can pass token path without changing relayer's code

Reply:
I would say that we need to be flexible for adding new tokens easily - not sure though whether we should prefer changing code or adding runtime possibility

Reply:
It seems invalid to me to not have all of the config for a token in place - like missing the contract address or missing the price feed address and so on - so I was thinking of pushing the checks as close to compile-time as possible. WDYT?

Reply:
Fair enough